### PR TITLE
documenting TELCODOCS-1357 SNO reboot behaviour

### PR DESCRIPTION
**Pull request by kquinn1204**
[TELCODOCS-1357]: SNO Restarting a node changes the CPU cores pinning in the pod

Version(s): 4.14, 4.13, 4.12 and main

Issue:https://issues.redhat.com/browse/TELCODOCS-1357 addresses https://issues.redhat.com/browse/OCPBUGS-2180

Link to docs preview:https://61221--docspreview.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-reference-cluster-configuration-for-vdu.html#ztp-sno-node-reboot-scenarios_sno-configure-for-vdu

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/ztp-sno-node-reboot-scenarios.adoc [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/ztp_far_edge/ztp-reference-cluster-configuration-for-vdu.adoc
+
+:_content-type: CONCEPT
+[id="ztp-sno-node-reboot-scenarios_{context}"]
+= SNO node reboot scenario
+
+On a {sno-caps} cluster and in {product-title} clusters generally a situation might arise due to a node reboot or kubelet restart where an associated device plugin restart has the affect of causing the application pod to fail with the `UnexpectedAdmissionError` error. This error is reported due to the fact that application pods that need devices come up before the pod that is serving the devices because there is no way to control the pod restart order.
```
Comment (swatisehgal) [inline modules/ztp-sno-node-reboot-scenarios.adoc:9]: We have identified failure at kubelet restarts as a regression and is being worked on right now so let's slip mentioning that.
```suggestion
On a {sno-caps} cluster and in {product-title} clusters generally a situation might arise in case a node reboot occurs without node drain where an application pod requesting devices fails with the `UnexpectedAdmissionError` error. This error is reported due to the fact that application pods that need devices can come up before the pod that is serving the devices because there is no way to control the pod restart order.
```

File: modules/ztp-sno-node-reboot-scenarios.adoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+While this behavior is expected, it can lead to a pod remaining on the cluster  even when it has failed to deploy successfully and it will continue to report `UnexpectedAdmissionError`. This issue might cause undue concern particularly in a {sno-caps} deployment, where the option to drain the node is unavailable.
```
Comment (swatisehgal) [inline modules/ztp-sno-node-reboot-scenarios.adoc:11]: Please make sure to check if it is okay to add an upstream link like I have in the text below before applying my suggestion below:

```suggestion
While this behavior is expected, it can lead to a pod remaining on the cluster even when it has failed to deploy successfully and will continue to report `UnexpectedAdmissionError`. This issue is however alleviated by the fact that application pods are typically wrapped up in a deployment, replicaset or a daemonset, ensuring that the subsequent pods are created and run successfully to ensure successful deployment of the application. 

There is ongoing work to ensure that such [pods are gracefully terminated](https://github.com/kubernetes/kubernetes/issues/118472) but while that materializes a simple command as below can be used to remove the failed pods:
`
kubectl delete pods --field-selector status.phase=Failed -n <POD_NAMESPACE>
`
This issue might cause undue concern particularly in a {sno-caps} deployment, where the option to drain the node is unavailable.
```

File: modules/ztp-sno-node-reboot-scenarios.adoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+In multi-node deployments, you drain a node, which results in the transfer of workloads to other worker nodes in the deployment.
```
Comment (swatisehgal) [inline modules/ztp-sno-node-reboot-scenarios.adoc:12]: Can we add a link to documentation with details on how to drain the nodes here. Maybe [this](https://docs.openshift.com/container-platform/4.13/nodes/nodes/nodes-nodes-working.html) one?  I am guessing we'd have to update the version depending on the targeted version?

**Review by swatisehgal — COMMENTED** [review]
Thanks for working on this. Appreciate it!

I have provided a few suggestions as part of my review.

Adding other members from the compute team for review
/cc @ffromani @MarSik

**Review by tmalove — APPROVED** [review]

**Comment by tmalove** [comment]
/remove-label peer-review-in-progress
/label peer-review-done
@kquinn1204 Be sure to squash your commits before a merge request. Thanks!

**Comment by kquinn1204** [comment]
@swatisehgal @ffromani @MarSik any final comments before I push for a merge into the docs. Add a LGTM if happy I added small line ongoing upstream work without link to the specific issue (as not really allowed to do this). Hopefully line is OK as I have not promised this will be addressed. WDYT ?

**Review by ffromani — COMMENTED** [review]
/lgtm

The content is technically correct.

A possible further clarification could be something along the line of reminding the audience that if the workload is managed by a deployment (or daemonset or any higher level controller) having a pod in failed state is a very little concern if another instance is running, as it is expected to happen in this case.

This is because purpose of these controllers is exactly to have self-healing workloads and to react to failed pods sending others to the cluster.

In other words, albeit suboptimal, this case is fully handled by the core features of the platform and should pause no concern.

I concur that a periodic cleanup of some form is still warranted to free the very little cluster  resources consumed by failed pods.

All the above is meant to be a suggestion and is optional. Totally not blocking, feel free to skip it or incorporate partially or to defer to a followup.

**Comment by shajmakh** [comment]
@kquinn1204  Thanks for your PR.
LGTM
